### PR TITLE
UpdateClient

### PR DESCRIPTION
**Pull request by 2841156671**
UpdateClient

### CODE REVIEW

File: client/height.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	splitStr := strings.Split(chainID, "-")
+	revision, err := strconv.ParseUint(splitStr[len(splitStr)-1], 10, 64)
+	// sanity check: error should always be nil since regex only allows numbers in last element
+	if err != nil {
+		panic(fmt.Sprintf("regex allowed non-number value as last split element for chainID: %s", chainID))
+	}
+	return revision
+}
+
```
Comment (dreamer-zq) [inline client/height.go]: delete

File: client/height.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	var a, b big.Int
+	if h.RevisionNumber != height.RevisionNumber {
+		a.SetUint64(h.RevisionNumber)
+		b.SetUint64(height.RevisionNumber)
+	} else {
+		a.SetUint64(h.RevisionHeight)
+		b.SetUint64(height.RevisionHeight)
+	}
+	return int64(a.Cmp(&b))
+}
```
Comment (dreamer-zq) [inline client/height.go:66]: delete

File: client/types.go [hunk 3]
```diff
@@ -28,9 +28,25 @@ func (ics IdentifiedClientStates) Sort() IdentifiedClientStates {
...
+	return "tibc"
+}
+
+func (m *MsgUpdateClient) Type() string {
+	return "update_client"
+}
+
+func (m *MsgUpdateClient) ValidateBasic() error {
+	return nil
 }
```
Comment (dreamer-zq) [inline client/types.go]: ?

**Review by chengwenxi — APPROVED** [review]

File: client/types.go [hunk 4]
```diff
@@ -28,9 +28,25 @@ func (ics IdentifiedClientStates) Sort() IdentifiedClientStates {
...
 
+func (m *MsgUpdateClient) GetSignBytes() []byte {
+	panic("IBC messages do not support amino")}
```
Comment (dreamer-zq) [inline client/types.go]: ?

File: integration_test/lightClient_test.go [hunk 5]
```diff
@@ -123,7 +124,7 @@ func getConsensusState(client clientforlightclient) {
 	if err != nil {
 		panic(err)
 	}
-
+	fmt.Println(consensusState1.GetTimestamp())
 	fmt.Println(consensusState1.String())
```
Comment (dreamer-zq) [inline integration_test/lightClient_test.go:128]: delete

File: tendermint/errors.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	moduleName    = "tibc" + "-" + SubModuleName
+)
+
+// IBC tendermint client sentinel errors
+var (
+	ErrInvalidChainID         = Register(moduleName, 2, "invalid chain-id")
+	ErrInvalidTrustingPeriod  = Register(moduleName, 3, "invalid trusting period")
+	ErrInvalidUnbondingPeriod = Register(moduleName, 4, "invalid unbonding period")
+	ErrInvalidHeaderHeight    = Register(moduleName, 5, "invalid header height")
+	ErrInvalidHeader          = Register(moduleName, 6, "invalid header")
```
Comment (dreamer-zq) [inline tendermint/errors.go:18]: delete errors.go file

**Review by dreamer-zq — COMMENTED** [review]
111

**Review by aofengli — APPROVED** [review]

**Pull request merged**